### PR TITLE
Small functional changes

### PR DESCRIPTION
**Pull request by httpsOmkar**
### Reasons for making this change

Some lint issues and missing hasOwnProperty while iterating over `for in`.

If this is related to existing tickets, include links to them as well.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [*] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by epicfaace** [comment]
Thanks! Can you update the title to be more specific as to what it does (the commit summary when merging will have the title name by default)

File: src/utils.js [hunk 1]
```diff
@@ -901,14 +901,16 @@ export function toPathSchema(schema, name = "", definitions, formData = {}) {
...
     for (const property in schema.properties) {
-      pathSchema[property] = toPathSchema(
-        schema.properties[property],
-        `${name}.${property}`,
-        definitions,
-        // It's possible that formData is not an object -- this can happen if an
-        // array item has just been added, but not populated with data yet
-        (formData || {})[property]
-      );
+      if (schema.properties.hasOwnProperty(property)) {
```
Comment (epicfaace) [inline src/utils.js]: Is there an example of a schema in which the original code would fail? Could you add a test with it?

Comment (httpsOmkar) [reply src/utils.js]: https://stackoverflow.com/questions/12735778/for-in-and-hasownproperty

**Comment by epicfaace** [comment]
@httpsOmkar it's been a while since this PR was updated, so I'm going to close it for now. Feel free to make a new PR as long as it 1) fixes a specific issue that is described in the PR description and 2) adds test cases that cover what you're fixing.

**Pull request closed without merging**